flang/test/Driver/save-temps.f90

```diff
@@ -0,0 +1,53 @@
+! Tests for the `-save-temps` flag. As `flang` does not implement `-fc1as` (i.e. a driver for the intergrated assembly), we need to
+! use `-fno-integrated-as` here.
+
+!--------------------------
+! Basic case: `-save-temps`
+!--------------------------
+! RUN: %flang -save-temps -fno-integrated-as %s -### 2>&1 | FileCheck %s
+! CHECK: "-o" "save-temps.i"
+! CHECK-NEXT: "-o" "save-temps.bc"
+! CHECK-NEXT: "-o" "save-temps.s"
+! CHECK-NEXT: "-o" "save-temps.o"
+! CHECK-NEXT: "-o" "a.out"
+
+!--------------------------
+! `-save-temps=cwd`
+!--------------------------
+! This should work the same as -save-temps above
+
+! RUN: %flang -save-temps=cwd -fno-integrated-as  %s -### 2>&1 | FileCheck %s -check-prefix=CWD
+! CWD: "-o" "save-temps.i"
+! CWD-NEXT: "-o" "save-temps.bc"
+! CWD-NEXT: "-o" "save-temps.s"
+! CWD-NEXT: "-o" "save-temps.o"
+! CWD-NEXT: "-o" "a.out"
+
+!--------------------------
+! `-save-temps=obj`
+!--------------------------
+! Check that temp files are saved in the same directory as the output file
+! regardless of whether -o is specified.
+
+! RUN: %flang -save-temps=obj -fno-integrated-as -o obj/dir/a.out %s -### 2>&1 | FileCheck %s -check-prefix=CHECK-OBJ
+! CHECK-OBJ: "-o" "obj/dir{{/|\\\\}}save-temps.i"
```

Comment:
Why does these checks test for `/` as the first directory separator and either `/` or `\\` as the second directory separator?

Reply:
TBH, I've just copied this [from Clang](https://github.com/llvm/llvm-project/blob/main/clang/test/Driver/save-temps.c#L66-L73) and forgot to check. I couldn't find any rationale in Clang, so I will just change this to plain "/".